assembly/src/parsers/blocks.rs

```diff
@@ -56,17 +60,34 @@ impl BlockParser {
         tokens: &mut TokenStream,
         context: &AssemblyContext,
         num_proc_locals: u32,
+        in_debug_mode: bool,
     ) -> Result<CodeBlock, AssemblyError> {
         match self {
             Self::Span => {
                 // --------------------------------------------------------------------------------
                 let mut span_ops = Vec::new();
                 let mut decorators = DecoratorList::new();
+                let mut ops_len = 0;
+                let mut dec_len = 0;
                 while let Some(op) = tokens.read() {
                     if op.is_control_token() {
                         break;
                     }
+                    // if assembler is in debug mode, populate decorators list with debug related
+                    // decorators like AsmOp.
+                    if in_debug_mode {
+                        dec_len = decorators.len();
+                        ops_len = span_ops.len();
+                        decorators
+                            .push((ops_len, Decorator::AsmOp(AsmOpInfo::new(op.to_string(), 1))));
+                    }
                     parse_op_token(op, &mut span_ops, num_proc_locals, &mut decorators)?;
+                    if in_debug_mode {
+                        // edit the number of cycles corresponding to the asmop decorator at an index
+                        if let Decorator::AsmOp(asmop_info) = &mut decorators[dec_len].1 {
+                            asmop_info.set_cycles((span_ops.len() - ops_len) as u8)
+                        }
+                    }
```

Comment:
would handling this inside `parse_op_token` be more difficult than here?

Reply:
Moved to parse_op_token

Reply:
@bobbinth - Sorry, did you mean to handle it inside individual parse methods to avoid 2 if blocks?

Reply:
No no - I think this is good. Btw, one thing that would be good to find out is time it takes to compile/run both in debug and regular mode, and compare this to what was before this PR.

Reply:
Mean compilation time before this PR: 48.26 ms
Mean compilation time after this PR without debug mode: 48.292 ms
Mean compilation time before this PR with debug mode: 50.753 ms

Reply:
@bobbinth - Would it be a good idea to have this as a soft requirement with any PR (post before and after benchmarks)?

Reply:
I think for PRs which affect the overall structure, this should definitely be the case going forward. But I'm not sure we should do this for all PRs. At least not yet.